rust/dialog-storage/src/storage/backend.rs

```diff
@@ -14,6 +14,14 @@ mod indexeddb;
 #[cfg(all(target_arch = "wasm32", target_os = "unknown"))]
 pub use indexeddb::*;
 
+/// S3/R2-compatible storage backend.
+pub mod s3;
+pub use s3::{S3, S3StorageError};
```

Comment:
Should we feature-gate this part of the crate? E.g., so that someone who doesn't care about remote storage doesn't incur the related dependencies.